dask/dataframe/io/parquet/core.py

```diff
@@ -95,6 +96,56 @@ def __call__(self, part):
         )
 
 
+class ToParquetFunctionWrapper:
+    """
+    Parquet Function-Wrapper Class
+
+    Writes a DataFrame partition into a distinct
+    parquet file. When called, the function also
+    requires a ``partition_info`` kwarg.
+    """
+
+    def __init__(
+        self,
+        engine,
+        path,
+        fs,
+        partition_on,
+        write_metadata_file,
+        i_offset,
+        name_function,
+        kwargs_pass,
+    ):
+        self.engine = engine
+        self.path = path
+        self.fs = fs
+        self.partition_on = partition_on
+        self.write_metadata_file = write_metadata_file
+        self.i_offset = i_offset
+        self.name_function = name_function
+        self.kwargs_pass = kwargs_pass
+
+    def __call__(self, df, block_index):
```

Comment:
```suggestion
    def __call__(self, df: pd.DataFrame, block_index: tuple[int]):
```

Reply:
`df` may also be a `cudf.DataFrame` - Is there an established way that we are dealing with this ambiguity in Dask?

Reply:
ooof, probably not! maybe just leave this off then. the `block_index` one was more interesting to me anyway.